aries_vcx/src/handlers/connection/util.rs

```diff
@@ -11,7 +11,7 @@ pub fn handle_ping(ping: &Ping,
 ) -> VcxResult<()> {
     if ping.response_requested {
         let ping_response = PingResponse::create().set_thread_id(
-            &ping.thread.as_ref().and_then(|thread| thread.thid.clone()).unwrap_or(ping.id.0.clone()));
+            &ping.thread.as_ref().and_then(|thread| Some(thread.thid.clone())).unwrap_or(ping.id.0.clone()));
```

Comment:
If you extract this statement into function like `build_ping_response(ping: &Ping) -> PingResponse`, it can be unit tested for both cases (when Ping contains `thread` and when it only contains `id`).

Reply:
Test added.